pkg/cache/resource_test.go

```diff
@@ -0,0 +1,275 @@
+			wantAvailable:            map[string]resources.FlavorResourceQuantities{"cq1": {{Flavor: "red", Resource: "cpu"}: 9_000}},
+			wantPotentiallyAvailable: map[string]resources.FlavorResourceQuantities{"cq1": {{Flavor: "red", Resource: "cpu"}: 20_000}},
+		},
+		"cq oversubscription spills into cohort": {
+			clusterQueues: []kueue.ClusterQueue{
+				utiltesting.MakeClusterQueue("cq1").
+					Cohort("cohort").
+					ResourceGroup(
+						*utiltesting.MakeFlavorQuotas("red").Resource("cpu", "10").Obj(),
+					).ClusterQueue,
+				utiltesting.MakeClusterQueue("cq2").
+					Cohort("cohort").
+					ResourceGroup(
+						*utiltesting.MakeFlavorQuotas("red").Resource("cpu", "10", "0", "0").Obj(),
+					).ClusterQueue,
+			},
+			cohorts: []kueuealpha.Cohort{
+				utiltesting.MakeCohort("cohort").
+					ResourceGroup(
+						*utiltesting.MakeFlavorQuotas("red").Resource("cpu", "10").Obj(),
+					).Cohort,
+			},
+			usage: map[string]resources.FlavorResourceQuantities{
+				"cq1": {{Flavor: "red", Resource: "cpu"}: 31_000},
+			},
+			wantAvailable: map[string]resources.FlavorResourceQuantities{
+				"cq1": {{Flavor: "red", Resource: "cpu"}: -11000},
+				"cq2": {{Flavor: "red", Resource: "cpu"}: -1000},
+			},
+			wantPotentiallyAvailable: map[string]resources.FlavorResourceQuantities{
+				"cq1": {{Flavor: "red", Resource: "cpu"}: 20_000},
+				"cq2": {{Flavor: "red", Resource: "cpu"}: 10_000},
+			},
+		},
+		"lending and borrowing limits respected": {
+			// cohort has 40 (red, cpu) total
+			// cq1 can access 33 = 40 - 7 reserved
+			// cq2 can access 12, due to borrowing limit
+			// cq3 can access 40, as there are no restrictions.
+			clusterQueues: []kueue.ClusterQueue{
+				utiltesting.MakeClusterQueue("cq1").
+					Cohort("cohort").
+					ResourceGroup(
+						*utiltesting.MakeFlavorQuotas("red").Resource("cpu", "10").Obj(),
+					).ClusterQueue,
+				utiltesting.MakeClusterQueue("cq2").
+					Cohort("cohort").
+					ResourceGroup(
+						*utiltesting.MakeFlavorQuotas("red").Resource("cpu", "10", "2").Obj(),
+					).ClusterQueue,
+				utiltesting.MakeClusterQueue("cq3").
+					Cohort("cohort").
+					ResourceGroup(
+						*utiltesting.MakeFlavorQuotas("red").Resource("cpu", "10", "", "3").Obj(),
+					).ClusterQueue,
+			},
+			cohorts: []kueuealpha.Cohort{
+				utiltesting.MakeCohort("cohort").
+					ResourceGroup(
+						*utiltesting.MakeFlavorQuotas("red").Resource("cpu", "10").Obj(),
+					).Cohort,
+			},
+			usage: map[string]resources.FlavorResourceQuantities{
+				"cq1": {{Flavor: "red", Resource: "cpu"}: 20_000},
+				"cq2": {{Flavor: "red", Resource: "cpu"}: 10_000},
+				"cq3": {{Flavor: "red", Resource: "cpu"}: 5_000},
+			},
+			wantAvailable: map[string]resources.FlavorResourceQuantities{
+				"cq1": {{Flavor: "red", Resource: "cpu"}: 3_000},
+				"cq2": {{Flavor: "red", Resource: "cpu"}: 2_000},
+				"cq3": {{Flavor: "red", Resource: "cpu"}: 5_000},
```

Comment:
I think each of these values require a comment

Reply:
Done. Also modified the numbers slightly to make the math more simple (3->5)